h2/src/main/org/h2/command/dml/Select.java

```diff
@@ -1060,14 +1060,22 @@ public String getPlanSQL() {
         StatementBuilder buff = new StatementBuilder();
         for (TableFilter f : topFilters) {
             Table t = f.getTable();
-            if (t.isView() && ((TableView) t).isRecursive()) {
-                buff.append("WITH RECURSIVE ").append(t.getName()).append('(');
-                buff.resetCount();
-                for (Column c : t.getColumns()) {
-                    buff.appendExceptFirst(",");
-                    buff.append(c.getName());
+            TableView tableView = (t instanceof TableView) ? (TableView) t : null;
```

Comment:
Why did you change `t.isView()` to `(t instanceof TableView)`?

Reply:
Reverted to isView() - I checked that no other subclasses of Table return true on isView calls.